source/dub/compilers/utils.d

```diff
@@ -326,31 +326,37 @@ NativePath generatePlatformProbeFile()
 	enum probe = q{
 		module object;
 
+		template _d_arrayappendcTXImpl(Tarr : T[], T)
```

Comment:
This can't go in.

It is a time bomb waiting to happen.

Get rid of the appends instead.

Reply:
why not? it's just for the platform probe and it will be _very_ apparent if it breaks

Reply:
did you add a test case for that to the CI? Otherwise it might get introduced again

Reply:
> why not? it's just for the platform probe and it will be _very_ apparent if it breaks - technically could wrap this in a version LDC/GDC or static if, when LDC and GDC fix this issue

Exactly, we'll know about it, There won't be a workaround and it limits what frontend versions dub can work for.

It is far too easy for this to fail, especially when compiler hooks are in flux.

Reply:
fixed, solution will can no longer break with newer compilers anymore and works with several tested older compilers, as well as CI should cover it.